### DISCUSSION TITLE
Reset field access in controller

### DISCUSSION THREAD

**Discussion by sanduluca** [react-hook-form/react-hook-form#9857] [Q&A]
Hi. I have an input like in the following image. I am using the Controller component to render my text input component. 
The question is: **How do I implement the reset of the field ?**

I know about the `resetField` on the `useForm` hook. Should I always extract it from useForm and pass it to my text input component? It would be great if the resetField would be accessible at controller level

React Native

![image](https://user-images.githubusercontent.com/56228534/216288843-9b5b3400-438f-4741-a5ae-889154a5981b.png)

**Comment by Moshyfawn** [comment]
Depending on what you expect to happen with the input on reset, you can simply use the `onChange` handler and pass it the default field value.

The `resetField` currently needs to be a part of `useForm` return. You can either pass it as prop or by `FormProvider` and get it via `useFormContext`

**Reply by sanduluca** [reply]
> and pass it the default field value.

Does this mean that I should pass `defaultValue` to `useForm` and to `useController` as well ? That's a little awkward. 


<br />

> The `resetField` currently needs to be a part of `useForm` return. You can either pass it as prop or by `FormProvider` and get it via `useFormContext`

How about having a reset function output in the controller, so that you can reset field value to default value used by the controller?

Currently I need to pass the `resetField` function from useForm to the input component. But it would be more convenient to get it directly from a controller so you can reset the field value.
